File: Progammers_위장.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> cloth_map;
    
	/* push into the map */
    for(auto cloth : clothes)
    {
       cloth_map[cloth[1]]++; 
    }
    
	/* get map elements by auto iter */
    for(auto iter = cloth_map.begin() ; iter != cloth_map.end(); iter++){
		answer *= (iter->second + 1); 
	}
    
    return answer - 1;
}

/* revised from https://rooted.tistory.com/14 */

File: Progammers_위장.py
def solution(clothes):
    answer = 1
    cloth_map = {}    # Dictionary
    
    for cloth in clothes:
        if cloth[1] in cloth_map:
            cloth_map[cloth[1]] += 1
        else:
            cloth_map[cloth[1]] = 1
    
    # only get the values of the map
    for types in cloth_map.values():
        answer *= (types + 1)
    
    return answer - 1